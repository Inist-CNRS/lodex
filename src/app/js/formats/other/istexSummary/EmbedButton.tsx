import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { IconButton } from '@mui/material';
import CodeIcon from '@mui/icons-material/Code';
import classnames from 'classnames';

import { polyglot as polyglotPropTypes } from '../../../propTypes';
import ButtonWithDialog from '../../../lib/components/ButtonWithDialog';
import { getCleanHost } from '../../../../../common/uris';

class EmbedButton extends Component {
    state = { open: false };

    handleOpen = () => this.setState({ open: true });
    handleClose = () => this.setState({ open: false });

    renderOpenButton = () => {
        // @ts-expect-error TS2339
        const { className, p: polyglot } = this.props;

        return (
            // @ts-expect-error TS2769
            <IconButton
                className={classnames('embed-button', className)}
                style={{ position: 'absolute' }}
                tooltip={polyglot.t('embed_istex_summary')}
                onClick={this.handleOpen}
            >
                <CodeIcon />
            </IconButton>
        );
    };

    renderDialog = () => {
        // @ts-expect-error TS2339
        const { uri, fieldName, p: polyglot } = this.props;
        const host = getCleanHost();

        return (
            <>
                <p>{polyglot.t('embed_step_head')}</p>
                <pre>
                    {`<script src="${host}/embeddedIstexSummary.js" defer></script>`}
                </pre>
                <p>{polyglot.t('embed_step_div')}</p>
                <pre>
                    {`<div class="embedded-istex-summary" data-api="${host}" data-uri="${uri}" data-field-name="${fieldName}"></div>`}
                </pre>
            </>
        );
    };

    render() {
        // @ts-expect-error TS2339
        const { p: polyglot } = this.props;
        const { open } = this.state;

        return (
            <ButtonWithDialog
                label={polyglot.t('embed_istex_summary')}
                open={open}
                handleOpen={this.handleOpen}
                handleClose={this.handleClose}
                openButton={this.renderOpenButton()}
                dialog={this.renderDialog()}
            />
        );
    }
}

// @ts-expect-error TS2339
EmbedButton.propTypes = {
    className: PropTypes.string,
    p: polyglotPropTypes.isRequired,
    uri: PropTypes.string.isRequired,
    fieldName: PropTypes.string.isRequired,
};

// @ts-expect-error TS2339
EmbedButton.defaultProps = {
    className: null,
};

export default EmbedButton;
