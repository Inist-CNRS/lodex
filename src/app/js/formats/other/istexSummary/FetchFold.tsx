import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Folder from '@mui/icons-material/Folder';
import FolderOpen from '@mui/icons-material/FolderOpen';
import Arrow from '@mui/icons-material/KeyboardArrowDown';
import { Button, CircularProgress } from '@mui/material';
import get from 'lodash/get';

import { polyglot as polyglotPropTypes } from '../../../propTypes';
import AdminOnlyAlert from '../../../lib/components/AdminOnlyAlert';
import SkipFold from './SkipFold';
import stylesToClassname from '../../../lib/stylesToClassName';

const styles = stylesToClassname(
    {
        li: {
            listStyleType: 'none',
        },
        buttonLabel: {
            display: 'flex',
            alignItems: 'center',
            padding: '0px 8px',
        },
        labelText: {
            marginLeft: 8,
        },
        count: {
            marginLeft: 8,
            backgroundColor: '#EEE',
            borderRadius: '0.7em',
            fontSize: '0.7em',
            lineHeight: '0.7em',
            padding: '0.5em',
            fontStyle: 'italic',
        },
        arrowClose: {
            transform: 'rotate(-90deg)',
        },
    },
    'fetch-fold',
);

const circularProgress = (
    <CircularProgress
        variant="indeterminate"
        size={20}
        // @ts-expect-error TS2322
        innerStyle={{
            display: 'flex',
            marginLeft: 8,
        }}
    />
);

class FetchFold extends Component {
    state = {
        data: null,
        error: null,
        isLoading: false,
        isOpen: false,
    };

    open = () => {
        if (this.state.data) {
            this.setState({
                isOpen: true,
            });
            return;
        }

        this.setState({ isLoading: true }, () => {
            this.props
                // @ts-expect-error TS2339
                .getData(this.props)
                // @ts-expect-error TS7006
                .then((data) => {
                    this.setState({
                        data,
                        isLoading: false,
                        isOpen: true,
                    });
                })
                // @ts-expect-error TS7006
                .catch((error) => {
                    console.error(error);
                    this.setState({ error: true });
                });
        });
    };

    close = () => {
        this.setState({ isOpen: false });
    };

    render() {
        // @ts-expect-error TS2339
        const { label, count, polyglot, children, skip } = this.props;
        const { error, data, isOpen, isLoading } = this.state;

        if (skip) {
            return <SkipFold {...this.props} />;
        }

        if (error) {
            return <AdminOnlyAlert>{polyglot.t('istex_error')}</AdminOnlyAlert>;
        }
        if (count === 0) {
            return null;
        }

        return (
            <div className="istex-fold">
                <div>
                    <Button
                        // @ts-expect-error TS2769
                        color="text"
                        onClick={isOpen ? this.close : this.open}
                    >
                        {/*
                         // @ts-expect-error TS2339 */}
                        <div className={styles.buttonLabel}>
                            <Arrow
                                className={
                                    // @ts-expect-error TS2339
                                    isOpen ? undefined : styles.arrowClose
                                }
                            />
                            {isOpen ? <FolderOpen /> : <Folder />}
                            {/*
                             // @ts-expect-error TS2339 */}
                            <span className={styles.labelText}>{label}</span>
                            {/*
                             // @ts-expect-error TS2339 */}
                            <span className={styles.count}>{count}</span>
                            {isLoading && circularProgress}
                        </div>
                    </Button>
                    {isOpen &&
                        // @ts-expect-error TS2349
                        children({
                            ...this.props,
                            data,
                            nbSiblings: get(data, 'hits.length', 0),
                        })}
                </div>
            </div>
        );
    }
}

// @ts-expect-error TS2339
FetchFold.propTypes = {
    label: PropTypes.string.isRequired,
    polyglot: polyglotPropTypes.isRequired,
    getData: PropTypes.func.isRequired,
    children: PropTypes.func.isRequired,
    count: PropTypes.number.isRequired,
    skip: PropTypes.number,
};

export default FetchFold;
