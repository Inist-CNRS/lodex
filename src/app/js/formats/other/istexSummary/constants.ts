export const SORT_YEAR_DESC = 'YEAR_DESC';
export const SORT_YEAR_ASC = 'YEAR_ASC';

export const SORT_YEAR_VALUES = [SORT_YEAR_DESC, SORT_YEAR_ASC];

export const CUSTOM_ISTEX_QUERY = 'custom_istex_query';
export const HOST_ISSN = 'host.issn';
export const HOST_EISSN = 'host.eissn';
export const HOST_ISBN = 'host.isbn';
export const HOST_EISBN = 'host.eisbn';
export const HOST_TITLE = 'host.title';
export const HOST_TITLE_RAW = 'host.title.raw';
export const REFBIBS_HOST_TITLE_RAW = 'refBibs.host.title.raw';
export const REFBIBS_TITLE = 'refBibs.title';
export const TOP_HITS = 'topHits';

export const SEARCHED_FIELD_VALUES = [
    CUSTOM_ISTEX_QUERY,
    HOST_ISSN,
    HOST_EISSN,
    HOST_ISBN,
    HOST_EISBN,
    HOST_TITLE,
];
