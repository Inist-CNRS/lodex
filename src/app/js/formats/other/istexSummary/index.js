import translate from 'redux-polyglot/translate';

import Component from './IstexSummaryView';
import ListComponent from './IstexSummaryList';
import AdminComponent, { defaultArgs } from './IstexSummaryAdmin';
import DefaultFormat from '../../utils/components/default-format';

export default {
    ...DefaultFormat,
    Component: translate(Component),
    ListComponent,
    AdminComponent,
    defaultArgs,
};
