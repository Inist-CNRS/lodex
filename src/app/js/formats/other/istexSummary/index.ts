import { translate } from '../../../i18n/I18NContext';

import Component from './IstexSummaryView';
import ListComponent from './IstexSummaryList';
import AdminComponent, { defaultArgs } from './IstexSummaryAdmin';
import DefaultFormat from '../../utils/components/default-format';

export default {
    ...DefaultFormat,
    Component: translate(Component),
    ListComponent,
    AdminComponent,
    defaultArgs,
};
