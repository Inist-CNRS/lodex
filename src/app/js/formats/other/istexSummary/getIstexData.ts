import get from 'lodash/get';
import omit from 'lodash/omit';

// @ts-expect-error TS7016
import composeAsync from '../../../../../common/lib/composeAsync';
import { parseFetchResult, output } from '../../utils/fetchIstexData';
// @ts-expect-error TS7016
import { ISTEX_API_URL } from '../../../../../common/externals';
import fetch from '../../../lib/fetch';
import alphabeticalSort from '../../../lib/alphabeticalSort';
import { CUSTOM_ISTEX_QUERY, SORT_YEAR_DESC } from './constants';

const defaultQueryOptions = {
    query: '',
    facet: null,
    output: '*',
    size: 0,
};

export const buildIstexQuery = (options = defaultQueryOptions) => {
    const opts = { ...defaultQueryOptions, ...options };
    const params = {
        ...omit(opts, ['query']),
        q: `(${opts.query})`,
    };

    const queryParams = Object.entries(params)
        .filter((param) => Number.isInteger(param[1]) || !!param[1])
        .map(
            ([key, value]) =>
                // @ts-expect-error TS2345
                `${encodeURIComponent(key)}=${encodeURIComponent(value)}`,
        )
        .join('&');

    return `${ISTEX_API_URL}/document/?${queryParams}`;
};

// @ts-expect-error TS7006
export const getFilterQuery = (searchedField, value) =>
    searchedField === CUSTOM_ISTEX_QUERY
        ? `(${value})`
        : `${searchedField}:"${value}"`;

// @ts-expect-error TS7031
export const getYearUrl = ({ resource, field, searchedField }) => {
    const value = resource[field.name];

    if (!value || !searchedField) {
        return null;
    }

    return buildIstexQuery({
        query: getFilterQuery(searchedField, value),
        // @ts-expect-error TS2322
        facet: 'publicationDate[perYear]',
    });
};

export const getDecadeYearUrl =
    // @ts-expect-error TS7031


        ({ value, to, from, searchedField }) =>
        () => ({
            url: buildIstexQuery({
                query: `${getFilterQuery(
                    searchedField,
                    value,
                )} AND publicationDate:[${from} TO ${to}]`,
                // @ts-expect-error TS2322
                facet: 'publicationDate[*-*:1]',
            }),
        });

export const getDecadeYearData = ({
    // @ts-expect-error TS7031
    value,
    // @ts-expect-error TS7031
    to,
    // @ts-expect-error TS7031
    from,
    // @ts-expect-error TS7031
    searchedField,
    // @ts-expect-error TS7031
    sortDir,
}) =>
    composeAsync(
        getDecadeYearUrl({ value, to, from, searchedField }),
        fetch,
        // @ts-expect-error TS2339
        parseFacetData('publicationDate', ({ keyAsString }) => keyAsString),
        // @ts-expect-error TS7006
        (data) => ({
            ...data,
            // @ts-expect-error TS7006
            hits: data.hits.sort((a, b) =>
                sortDir === SORT_YEAR_DESC ? b.name - a.name : a.name - b.name,
            ),
        }),
    );

// @ts-expect-error TS7006
export const parseYearData = (formatData, sortDir = SORT_YEAR_DESC) => ({
    hits: get(formatData, 'aggregations.publicationDate.buckets', [])
        // @ts-expect-error TS7006
        .sort((a, b) =>
            sortDir === SORT_YEAR_DESC
                ? b.keyAsString - a.keyAsString
                : a.keyAsString - b.keyAsString,
        )
        // @ts-expect-error TS7031
        .map(({ keyAsString, docCount }) => ({
            name: keyAsString,
            count: docCount,
        })),
});

export const getVolumeUrl =
    // @ts-expect-error TS7031


        ({ value, year, searchedField }) =>
        () => ({
            url: buildIstexQuery({
                query: `${getFilterQuery(
                    searchedField,
                    value,
                )} AND publicationDate:"${year}"`,
                // @ts-expect-error TS2322
                facet: 'host.volume[*-*:1]',
            }),
        });

export const parseFacetData =
    // @ts-expect-error TS7006


        (facetName, getName = ({ key }) => key) =>
        // @ts-expect-error TS7031
        ({ response, error }) => {
            if (error) {
                throw error;
            }

            return {
                hits: get(
                    response,
                    ['aggregations', facetName, 'buckets'],
                    [],
                ).map(
                    // @ts-expect-error TS7031
                    ({ docCount, ...data }) => ({
                        // @ts-expect-error TS2345
                        name: getName(data),
                        count: docCount,
                    }),
                ),
            };
        };

export const parseVolumeData = parseFacetData('host.volume');

export const getOtherVolumeUrl =
    // @ts-expect-error TS7031


        ({ value, year, searchedField }) =>
        () => ({
            url: buildIstexQuery({
                query: `${getFilterQuery(
                    searchedField,
                    value,
                )} AND publicationDate:"${year}" AND -host.volume:[0 TO *]`,
                output: 'host.volume',
                // @ts-expect-error TS2322
                size: '*',
            }),
        });

export const parseOtherData =
    // @ts-expect-error TS7006


        (key) =>
        // @ts-expect-error TS7031
        ({ response, error }) => {
            if (error) {
                throw error;
            }
            // @ts-expect-error TS7006
            const count = response.hits.reduce((acc, hit) => {
                const name = get(hit, key, 'other');
                return {
                    ...acc,
                    [name]: get(acc, name, 0) + 1,
                };
            }, {});

            return Object.keys(count).map((name) => ({
                name,
                count: count[name],
            }));
        };

// @ts-expect-error TS7031
export const getOtherVolumeData = ({ value, year, searchedField }) =>
    composeAsync(
        getOtherVolumeUrl({ value, year, searchedField }),
        fetch,
        parseOtherData('host.volume'),
    );

export const addOtherVolumeData =
    // @ts-expect-error TS7031


        ({ value, year, searchedField }) =>
        // @ts-expect-error TS7031
        async ({ hits }) => ({
            hits: alphabeticalSort([
                ...hits,
                ...(await getOtherVolumeData({ value, year, searchedField })()),
            ]),
        });

// @ts-expect-error TS7031
export const getVolumeData = ({ value, year, searchedField }) =>
    composeAsync(
        getVolumeUrl({ value, year, searchedField }),
        fetch,
        parseVolumeData,
        addOtherVolumeData({ value, year, searchedField }),
    );

// @ts-expect-error TS7006
const getVolumeQuery = (volume) =>
    volume === 'other' ? '-host.volume.raw:*' : `host.volume.raw:"${volume}"`;

export const getIssueUrl =
    // @ts-expect-error TS7031


        ({ value, year, volume, searchedField }) =>
        () => ({
            url: buildIstexQuery({
                query: `${getFilterQuery(
                    searchedField,
                    value,
                )} AND publicationDate:"${year}" AND ${getVolumeQuery(volume)}`,
                // @ts-expect-error TS2322
                facet: 'host.issue[*-*:1]',
            }),
        });

export const parseIssueData = parseFacetData('host.issue');

export const getOtherIssueUrl =
    // @ts-expect-error TS7031


        ({ value, year, volume, searchedField }) =>
        () => ({
            url: buildIstexQuery({
                query: `${getFilterQuery(
                    searchedField,
                    value,
                )} AND publicationDate:"${year}" AND ${getVolumeQuery(
                    volume,
                )} AND -host.issue:[0 TO *]`,
                // @ts-expect-error TS2322
                size: '*',
                output: 'host.issue',
            }),
        });

// @ts-expect-error TS7031
export const getOtherIssueData = ({ value, year, volume, searchedField }) =>
    composeAsync(
        getOtherIssueUrl({ value, year, volume, searchedField }),
        fetch,
        parseOtherData('host.issue'),
    );

export const addOtherIssueData =
    // @ts-expect-error TS7031


        ({ value, year, volume, searchedField }) =>
        // @ts-expect-error TS7031
        async ({ hits }) => ({
            hits: alphabeticalSort([
                ...hits,
                ...(await getOtherIssueData({
                    value,
                    year,
                    volume,
                    searchedField,
                })()),
            ]),
        });

// @ts-expect-error TS7031
export const getIssueData = ({ value, year, volume, searchedField }) =>
    composeAsync(
        getIssueUrl({ value, year, volume, searchedField }),
        fetch,
        parseIssueData,
        addOtherIssueData({ value, year, volume, searchedField }),
    );

// @ts-expect-error TS7006
const getIssueQuery = (issue) =>
    issue === 'other' ? '-host.issue.raw:*' : `host.issue.raw:"${issue}"`;

export const getDocumentUrl =
    // @ts-expect-error TS7031


        ({ value, year, volume, issue, searchedField, documentSortBy }) =>
        () => ({
            url: buildIstexQuery({
                query: `${getFilterQuery(
                    searchedField,
                    value,
                )} AND publicationDate:"${year}" AND ${getVolumeQuery(
                    volume,
                )} AND ${getIssueQuery(issue)}`,
                output,
                // @ts-expect-error TS2353
                sortBy: documentSortBy,
                size: 10,
            }),
        });

export const getDocumentData = ({
    // @ts-expect-error TS7031
    value,
    // @ts-expect-error TS7031
    year,
    // @ts-expect-error TS7031
    volume,
    // @ts-expect-error TS7031
    issue,
    // @ts-expect-error TS7031
    searchedField,
    // @ts-expect-error TS7031
    documentSortBy,
}) =>
    composeAsync(
        getDocumentUrl({
            value,
            year,
            volume,
            issue,
            searchedField,
            documentSortBy,
        }),
        fetch,
        parseFetchResult,
    );

// @ts-expect-error TS7006
export const getMoreDocumentUrl = (nextPageURI) => ({
    url: nextPageURI,
});

// @ts-expect-error TS7006
export const getMoreDocumentData = (nextPageURI) =>
    composeAsync(getMoreDocumentUrl, fetch, parseFetchResult)(nextPageURI);
