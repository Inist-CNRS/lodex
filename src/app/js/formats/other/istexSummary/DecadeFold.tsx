import React from 'react';
import PropTypes from 'prop-types';

import FetchFold from './FetchFold';
import { getDecadeYearData } from './getIstexData';
import { SEARCHED_FIELD_VALUES, SORT_YEAR_VALUES } from './constants';
import { polyglot as polyglotPropTypes } from '../../../propTypes';

const DecadeFold = ({
    // @ts-expect-error TS7031
    value,
    item: {
        // @ts-expect-error TS7031
        name: { from, to },
        // @ts-expect-error TS7031
        count,
    },
    // @ts-expect-error TS7031
    searchedField,
    // @ts-expect-error TS7031
    sortDir,
    // @ts-expect-error TS7031
    polyglot,
    // @ts-expect-error TS7031
    children,
}) => (
    <FetchFold
        // @ts-expect-error TS2769
        label={`${from}-${to}`}
        count={count}
        from={from}
        to={to}
        polyglot={polyglot}
        getData={getDecadeYearData({
            value,
            from,
            to,
            searchedField,
            sortDir,
        })}
    >
        {children}
    </FetchFold>
);

DecadeFold.propTypes = {
    value: PropTypes.string.isRequired,
    item: PropTypes.shape({
        name: PropTypes.string.isRequired,
        count: PropTypes.number.isRequired,
    }).isRequired,
    searchedField: PropTypes.oneOf(SEARCHED_FIELD_VALUES),
    sortDir: PropTypes.oneOf(SORT_YEAR_VALUES),
    children: PropTypes.func.isRequired,
    polyglot: polyglotPropTypes.isRequired,
};

export default DecadeFold;
