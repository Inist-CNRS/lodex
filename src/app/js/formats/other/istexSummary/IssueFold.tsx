// @ts-expect-error TS6133
import React from 'react';
import PropTypes from 'prop-types';

import FetchFold from './FetchFold';
import { getDocumentData } from './getIstexData';
import { SEARCHED_FIELD_VALUES } from './constants';
import { polyglot as polyglotPropTypes } from '../../../propTypes';

const IssueFold = ({
    // @ts-expect-error TS7031
    item: { name: issue, count },
    // @ts-expect-error TS7031
    value,
    // @ts-expect-error TS7031
    year,
    // @ts-expect-error TS7031
    volume,
    // @ts-expect-error TS7031
    searchedField,
    // @ts-expect-error TS7031
    polyglot,
    // @ts-expect-error TS7031
    children,
    // @ts-expect-error TS7031
    documentSortBy,
}) => (
    <FetchFold
        // @ts-expect-error TS2769
        label={
            issue === 'other'
                ? polyglot.t('other_issue')
                : `${polyglot.t('issue')}: ${issue}`
        }
        skip={issue === 'other'}
        count={count}
        issue={issue}
        polyglot={polyglot}
        getData={getDocumentData({
            value,
            year,
            volume,
            issue,
            searchedField,
            documentSortBy,
        })}
    >
        {children}
    </FetchFold>
);

IssueFold.propTypes = {
    item: PropTypes.shape({
        name: PropTypes.string.isRequired,
        count: PropTypes.number.isRequired,
    }).isRequired,
    value: PropTypes.string.isRequired,
    year: PropTypes.string.isRequired,
    volume: PropTypes.string.isRequired,
    searchedField: PropTypes.oneOf(SEARCHED_FIELD_VALUES),
    children: PropTypes.func.isRequired,
    isOther: PropTypes.bool,
    polyglot: polyglotPropTypes.isRequired,
    documentSortBy: PropTypes.string.isRequired,
};

export default IssueFold;
