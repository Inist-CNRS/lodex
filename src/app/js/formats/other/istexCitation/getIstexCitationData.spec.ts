import {
    getCitationDocumentUrl,
    getCitationUrl,
    parseCitationData,
} from './getIstexCitationData';

import { CUSTOM_ISTEX_QUERY } from '../istexSummary/constants';

describe('getIstexCitationData', () => {
    describe('getCitationUrl', () => {
        it('should return url to get citation facet', () => {
            expect(
                getCitationUrl({
                    resource: {
                        value: 'refBibs.host.title:"The Lancet"',
                    },
                    field: {
                        name: 'value',
                    },
                    searchedField: 'searched.field',
                }),
            ).toMatchSnapshot();
        });

        it('should return the value if the custom ISTEX query is selected', () => {
            const query = 'CUSTOM_QUERY';

            const result = getCitationUrl({
                resource: { value: query },
                field: { name: 'value' },
                searchedField: CUSTOM_ISTEX_QUERY,
            });

            expect(result).toMatchSnapshot();
        });
    });

    describe('parseCitationData', () => {
        it('should extract journal name from data', () => {
            expect(
                parseCitationData({
                    aggregations: {
                        'host.title': {
                            buckets: [
                                { key: 'The Lancet', docCount: 9908 },
                                {
                                    key: 'Social Science & Medicine',
                                    docCount: 322,
                                },
                                {
                                    key: 'Behavioral and Brain Sciences',
                                    docCount: 178,
                                },
                                {
                                    key: 'Health Policy and Planning',
                                    docCount: 113,
                                },
                            ],
                        },
                    },
                }),
            ).toEqual({
                hits: [
                    { name: 'The Lancet', count: 9908 },
                    {
                        name: 'Social Science & Medicine',
                        count: 322,
                    },
                    {
                        name: 'Behavioral and Brain Sciences',
                        count: 178,
                    },
                    {
                        name: 'Health Policy and Planning',
                        count: 113,
                    },
                ],
            });
        });

        it('should return empty array if empty response', () => {
            expect(parseCitationData({})).toEqual({ hits: [] });
        });
    });

    describe('getCitationDocumentUrl', () => {
        it('should return url to get document', () => {
            expect(
                getCitationDocumentUrl({
                    value: 'refBibs.host.title:"The Lancet"',
                    name: 'journal',
                    searchedField: 'searched.field',
                    documentSortBy: 'publicationDate[desc]',
                })(),
            ).toMatchSnapshot();
        });
    });
});
