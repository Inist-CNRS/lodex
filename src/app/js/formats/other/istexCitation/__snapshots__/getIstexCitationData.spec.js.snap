// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getIstexCitationData getCitationDocumentUrl should return url to get document 1`] = `
{
  "url": "https://api.istex.fr/document/?output=id%2CarkIstex%2Ctitle%2CpublicationDate%2Cauthor%2Chost.genre%2Chost.title%2Chost.pages.first%2Chost.pages.last%2Chost.volume%2Chost.issue&size=10&sortBy=publicationDate%5Bdesc%5D&q=(searched.field%3A%22refBibs.host.title%3A%22The%20Lancet%22%22%20AND%20host.title.raw%3A%22journal%22)",
}
`;

exports[`getIstexCitationData getCitationUrl should return the value if the custom ISTEX query is selected 1`] = `"https://api.istex.fr/document/?facet=host.title%5B10%5D&output=id%2CarkIstex%2Ctitle%2CpublicationDate%2Cauthor%2Chost.genre%2Chost.title%2Chost.pages.first%2Chost.pages.last%2Chost.volume%2Chost.issue&size=10&q=((CUSTOM_QUERY))"`;

exports[`getIstexCitationData getCitationUrl should return url to get citation facet 1`] = `"https://api.istex.fr/document/?facet=host.title%5B10%5D&output=id%2CarkIstex%2Ctitle%2CpublicationDate%2Cauthor%2Chost.genre%2Chost.title%2Chost.pages.first%2Chost.pages.last%2Chost.volume%2Chost.issue&size=10&q=(searched.field%3A%22refBibs.host.title%3A%22The%20Lancet%22%22)"`;
